### PR TITLE
docs(lists): finally revamp readme

### PR DESCRIPTION
**Pull request by ajhalili2006**
## What's changed?

Simply put, just updated the README for the `lists` directory with some docs and useful stuff. Also, I also updated the Danger JS workflow to be able to work with the Dependabot PRs. No need for changelog updates for now.

### CODE REVIEW

File: .github/workflows/pr-reviews.yml [hunk 1]
```diff
@@ -1,21 +1,51 @@
...
+      labeled,
+      unlabeled,
+      edited,
+      opened,
+      synchronize,
+      locked,
+      unlocked,
+      review_requested,
+      review_request_removed
+    ]
```
Comment (ThePinsTeam) [inline .github/workflows/pr-reviews.yml:18]: The rendering looks kinda sus from GitHub, how about this?

```suggestion
on: 
  # Chaos ahead, PROCEED AT YOUR OWN RISK!
  pull_request_target:
    types: 
      - assigned
      - unassigned
      - labeled
      - unlabeled
      - edited
      - opened
      - synchronize
      - locked
      - unlocked
      - review_requested
      - review_request_removed
```

**Review by ThePinsTeam — CHANGES_REQUESTED** [review]
There's something that needs to be fixed before merging before the chaos happen.

**Review by ThePinsTeam — APPROVED** [review]
Nice, :shipit:!

**Comment by ajhalili2006** [comment]
Since `Require linear commits` is enabled on the main branch, we need to either rebase or squash, but I'll go to the squashing instead. YOLO.

**Pull request merged**